ia-wishlist-bot/adding_wishlist_ol.py

```diff
@@ -0,0 +1,132 @@
+               "Spanish", "2002", "893562252", "9706664998", "9789706664990"]
+        book = parse_wishlist_csv_row_to_dict(csv)
+        expected_url = None
+
+        self.assertTrue(expected_url == get_bookcover(book))
+
+
+def process_csv(filename):
+    """This function takes a csv file which was output from our whatever process created e.g. *new_wishlist_salman_1000.csv* and converts it into a python dictionary
+
+    Usage:
+        >>> parse_wishlist_csv_row_to_dict("foo,bar,baz,qux")
+        { "author": "foo", "title": "bar", ...}
+    """
+    with open(filename, mode='r') as infile:
+        reader = csv.reader(infile)
+
+        book_data = [row for row in reader]
+
+        return book_data
+
+def parse_wishlist_csv_row_to_dict(csv):
+    """This function takes a csv row which was output from our whatever process created e.g. *new_wishlist_salman_1000.csv* and converts it into a python dictionary
+
+    Usage:
+        >>> parse_wishlist_csv_row_to_dict("foo,bar,baz,qux")
+        { "author": "foo", "title": "bar", ...}
+    """
+    book = {"title": csv[0], "authors": ast.literal_eval(csv[1]), "language": csv[2], "date": csv[3], "oclc": csv[4], "isbn10": csv[5], "isbn13": csv[6]}
+    return book
+
+
+def get_author_object(author_name, author_birth_date=None, author_death_date=None):
+    """This takes an author name which was output from our CSV row and then either retrieves an ol Object which was already created or creates a new OL Object
+
+    Usage:
+        >>> get_author_object('Dan Brown')
+    """
+    author_olid = ol.Author.get_olid_by_name(author_name)
+    if author_olid:
+        return ol.get(author_olid)
+    else:
+        return common.Author(name=author_name)
+
+def get_bookcover(book):
+    url = "https://images.betterworldbooks.com/" + book.get('isbn10')[0:3] + "/" + book.get('isbn13') + ".jpg"
+    
+    r =requests.get(url)
+
+    if r.status_code == 200:
+        return url
+    return None
+
+def add_book_via_olclient(book, author_list, bookcover=None):
+
+    # Define a Book Object
+    new_book = common.Book(title=book.get("title"), authors=author_list, publish_date=book.get("date"), language=book.get("language"))
+
+
+    # Add metadata like ISBN 10 and ISBN 13
+    new_book.add_id(u'isbn_10', book.get("isbn10"))
+    new_book.add_id(u'isbn_13', book.get('isbn13'))
+    new_book.add_id(u'oclc', book.get('oclc'))
+
+    print(new_book)
+    newer_book = ol.create_book(new_book)
+
+    if bookcover:
+        newer_book.add_bookcover(bookcover)
+
+def process_book(book):
+    # make sure we've normalized the author name (e.g. first last?)
+    author_list = []
+    for author_name in book.get('authors'):
+        author_list.append(get_author_object(author_name))
+
+    # Bookcover search, etc
+    bookcover = get_bookcover(book)
+
+    # Add book to Open Library via olclient
+    add_book_via_olclient(book, author_list, bookcover)
+
+if __name__ == "__main__":
+    # csv_row = sys.argv[1]
```

Comment:
if we did use this `csv_row` it would change the program so that it accepts 1 line of csv when we call it:

e.g. `python adding_wishlist_ol.py a,b,c,d` where `a,b,c,d` is actually a line from the csv

And in this way we can run the parallel command on the csv and have each line be processed in parallel (instead of looping serially over all the lines of the file)